src/main/java/io/jenkins/plugins/signpath/SignPathStepBase.java

```diff
@@ -122,15 +150,45 @@ protected String ensureNotNull(String input, String name) throws SignPathStepInv
         return input;
     }
 
-    protected String getWithGlobalConfig(String value, Function<SignPathPluginGlobalConfiguration, String> getter) {
-        if (value == null || value.isEmpty()) {
-            SignPathPluginGlobalConfiguration config = GlobalConfiguration.all().get(SignPathPluginGlobalConfiguration.class);
-            if (config != null) {
-                return getter.apply(config);
-            }
+    protected String getWithGlobalConfig(String stepLevelValue, Function<SignPathPluginGlobalConfiguration, String> globalValueGetter, String paramName) throws SignPathStepInvalidArgumentException {
+
+        SignPathPluginGlobalConfiguration config = GlobalConfiguration.all().get(SignPathPluginGlobalConfiguration.class);
+        if (config == null) {
+            throw new IllegalStateException("SignPathPluginGlobalConfiguration is not available. Ensure it is properly configured.");
+        }
+        String globalVal = globalValueGetter.apply(config);
+
+        if (globalVal == null || globalVal.isEmpty()) {
+            // global value is not set yet, fallback to the obsolete step-level value
```

Comment:
This function doesn't know whether the step-value is obsolete or not (you probably mean deprecated here right?). Maybe rephrase the comment?

Reply:
Right, fixed